web-registry/src/hooks/useBasketTokens.ts

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect } from 'react';
+
+import { Basket } from '@regen-network/api/lib/generated/regen/ecocredit/basket/v1/types';
+import { QueryBasketsResponse } from '@regen-network/api/lib/generated/regen/ecocredit/basket/v1/query';
+import { QueryBalanceResponse } from '@regen-network/api/lib/generated/cosmos/bank/v1beta1/query';
+
+import useQueryBaskets from './useQueryBaskets';
+import useQueryBalance from './useQueryBalance';
+
+export interface BasketTokens {
+  basket: Basket;
+  balance?: QueryBalanceResponse | undefined;
+  displayDenom: string;
+}
+
+export default function useBasketTokens(
+  address: string | undefined,
+): BasketTokens[] {
+  const baskets = useQueryBaskets();
```

Comment:
I think it's fine here, but I get a bit concerned about hooks calling other hooks... things can quickly spiral to a chain of side effects that's hard to manage. In general, I prefer hooks do one thing specifically and get external dependencies passed through arguments (ie `useBasketTokens(address, baskets)`)

No need to change this though, I don't think. And I love the use of hooks to DRY up code!

Reply:
Things can always get messy and difficult to maintain if there is no design behind it ;P Here the idea, at least initially, is to have a layer of primitive queries respecting the RegenAPI and the DTOs it returns, and above that, an abstraction layer of our app's business for handling and preparing the received data. Soon we will begin to review in depth issues of architecture and data management in the application.

Anyway I understand the intention of the dependency injection that you suggest, but not in this case I would say. Or do you think that the ultimate responsibility for preparing the BasketTokens data lies with the UI component that uses the hook?

Thanks, David!!